apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -0,0 +1,195 @@
+// limitations under the License.
+
+using System;
+using System.Linq;
+using Google.Api.Gax;
+
+// ReSharper disable UnusedParameter.Local
+
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// </summary>
+    public sealed class SpannerCommandTextBuilder
+    {
+        private static readonly string s_insertCommand = "INSERT";
+        private static readonly string s_insertUpdateCommand = "INSERTUPDATE";
+        private static readonly string s_updateCommand = "UPDATE";
+        private static readonly string s_deleteCommand = "DELETE";
+        private static readonly string s_selectCommand = "SELECT";
+
+        private string _targetTable;
+
+        /// <summary>
+        /// </summary>
+        /// <param name="commandText"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        public SpannerCommandTextBuilder(string commandText)
+        {
+            GaxPreconditions.CheckNotNullOrEmpty(commandText, nameof(commandText));
+            var commandSections = commandText.Split(' ');
+            if (commandSections.Length < 2)
+                throw new InvalidOperationException($"{commandText} is not a recognized Spanner command.");
+            var newBuilder = new SpannerCommandTextBuilder();
+            if (!TryParseCommand(this, s_deleteCommand, SpannerCommandType.Delete, commandSections)
+                && !TryParseCommand(this, s_updateCommand, SpannerCommandType.Update, commandSections)
+                && !TryParseCommand(this, s_insertCommand, SpannerCommandType.Insert, commandSections)
+                && !TryParseCommand(this, s_insertUpdateCommand, SpannerCommandType.InsertOrUpdate, commandSections))
+            {
+                if (!commandSections[0].ToUpper().StartsWith(s_selectCommand))
+                    throw new InvalidOperationException($"{commandText} is not a recognized Spanner command.");
+                newBuilder.CommandText = commandText;
+                newBuilder.SpannerCommandType = SpannerCommandType.Select;
+            }
+        }
+
+        /// <summary>
+        /// </summary>
+        private SpannerCommandTextBuilder()
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        public string CommandText { get; private set; }
+
+        /// <summary>
+        /// </summary>
+        public SpannerCommandType SpannerCommandType { get; private set; }
+
+        /// <summary>
+        /// </summary>
+        public string TargetTable
+        {
+            get { return _targetTable; }
+            private set
+            {
+                ValidateTable(value);
+                _targetTable = value;
+            }
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="databaseTableName"></param>
+        /// <returns></returns>
+        public static SpannerCommandTextBuilder CreateDeleteTextBuilder(string databaseTableName)
+        {
```

Comment:
The problem I have with the ctor is the sql query vs other commands.  sqlquery takes a string for the sql query.  The other commands take a table name.  With a ctor, it will not be as obvious whats going on because the name of the arg would be ambiguous.
With these static methods, its explicit what you are creating.